ipfx/epochs.py

```diff
@@ -51,23 +51,39 @@ def get_stability_vm_epoch(stim_start, hz):
     return stim_start-1-num_steps, stim_start-1
 
 
+def get_recording_epoch(response):
+    """
+    Detect response epoch defined as interval from start to the last non-zero value of the response
+
+    Returns
+    -------
+    start,end: int indices of the epoch
+    """
+
+    if len(np.flatnonzero(response)) == 0:
+        end_idx = 0
```

Comment:
This will return a range with one point where I would expect a range with no points or an exception.

Reply:
if both start and end indices =0, then the array slice will be empty. Thus, the current implementation will return no points, or am I wrong?

Reply:
@sgratiy Okay fine. I just read the function comment and thought it is a one point range.

Reply:
I'm still no convinced.

Let's assume response is flat zero.

We venture into the if branch and then return `0, 0`.

One of the callers is `current_clamp_sweep_qc_features`
 https://github.com/AllenInstitute/ipfx/blob/09fcd8e15be3b1fe77231f8f614161323dbf3963/ipfx/qc_feature_extractor.py#L291-L320

which throws away the starting index and only uses the ending index. This ending index is then passed into `get_last_vm_epoch` which returns a tuple where the first index is larger than the second which then results in `measure_vm` return `0, 0`.

If that is how it is supposed to be, well good. At least I would not expect that an empty slice has a property of 0.

PS: You can factor our the `np.flatnonzero(response)` so that it is only called once.

Reply:
This is a good point and it raises the valid question what to return from the function in case response is flat zero. As it stands now, the `current_clamp_sweep_qc_features` are only called for the "completed" sweeps and flat response will not satisfy it.  I am revamping `current_clamp_sweep_qc_features` now for related reason: when the stimulus is flat zero. So this code should improve.
There are two approaches as I can see: 
1. Check a number of conditions whether the sweeps has all proper recording, stimulus, not truncated early, etc. Fail sweeps  which do not satisfy those conditions and only determine qc features on the passing sweeps. 
2. Go right ahead into determing qc features `current_clamp_sweep_qc_features` , but then use lots of try-except blocks to handle issues with bad data.

Any preference?

Reply:
I would prefer 1. This is usually also the way we do things in MIES.

Reply:
This is my preference as well. I am working on implementing it now. 
Is there something you still need me to update in this PR?